packages/server/src/unstable-core-do-not-import/procedure.ts

```diff
@@ -70,5 +73,10 @@ export type inferProcedureInput<TProcedure extends AnyProcedure> =
 export type inferProcedureParams<TProcedure> = TProcedure extends AnyProcedure
   ? TProcedure['_def']
   : never;
+
 export type inferProcedureOutput<TProcedure> =
   inferProcedureParams<TProcedure>['_output_out'];
+
+export type inferProcedureBuilderContext<
+  TProcedureBuilder extends AnyProcedureBuilder,
+> = Parameters<Parameters<TProcedureBuilder['mutation']>[0]>[0]['ctx'];
```

Comment:
Looks weird to get the 'mutation' here directly. But it is what seemed to have worked correctly.